lib/assert.js

```diff
@@ -145,13 +145,7 @@ function _deepEqual(actual, expected, strict) {
   if (actual === expected) {
     return true;
   } else if (actual instanceof Buffer && expected instanceof Buffer) {
-    if (actual.length != expected.length) return false;
-
-    for (var i = 0; i < actual.length; i++) {
-      if (actual[i] !== expected[i]) return false;
-    }
-
-    return true;
+    return process.binding('buffer').compare(actual, expected) === 0;
```

Comment:
Can you cache the return value from process.binding()?  Calling it repeatedly is very wasteful.  Just treat it as you would require() statements.